ksql-rest-app/src/main/java/io/confluent/ksql/rest/client/KsqlRestClient.java

```diff
@@ -201,11 +203,37 @@ private Response makeGetRequest(String path) {
     public QueryStream(Response response) {
       this.response = response;
 
+      this.closed = false;
       this.objectMapper = new ObjectMapper();
-      this.responseScanner = new Scanner((InputStream) response.getEntity(), StandardCharsets.UTF_8.name());
+      InputStreamReader isr = new InputStreamReader((InputStream)response.getEntity(), StandardCharsets.UTF_8);
+      Object lk = this;
+      String errMsg = closedIllegalStateMsg("hasNext()");
+      this.responseScanner = new Scanner(new Readable() {
```

Comment:
lambda?

Reply:
For some reason I thought we had to support building on Java 7. I'll change it.